### DISCUSSION TITLE
Have u guys considered making a markdown exporter?

### DISCUSSION THREAD

**Discussion by YousufSSyed** [ReagentX/imessage-exporter#230] [Ideas]
I haven't used iMessage exporter but have been thinking about it. Markdown would be good for formatting messages in plain text and embedding images in them without a complex format like HTML. They could be imported to Obsidian and enjoy whatever features and plugins Obsidian has, with the exported messages. This would really get users like me to use it.
* Dates could be headers, allowing for linking to sections in other notes in markdown apps like Obsidian.
 * There could be an option for what header levels someone wants, H1 - H6.
* An option for if the sender and timestamp should be in the same line.
 * And if they should be headers and which level (H1 - H6), as regular lines, or bullet points.
* images can be embedded with the syntax `![[path/to/image.png]]`, with an option with or without exact file paths (some editors support embedding external images without file paths.
* An option for the width of images. ![[image.png|100]] would be 100 pixels wide.
* An option to have all the images embedded in the same line or different ones. Same line: ![[image.png|100]] ![[image.png|100]] ![[image.png|100]]
 * Different ones:
```markdown
![[image.png|100]]
![[image.png|100]]
![[image.png|100]]
```
I haven't really looked at the codebase, but If I were to make an markdown exporter myself, would it be in [the exporters folder](https://github.com/ReagentX/imessage-exporter/tree/develop/imessage-exporter/src/exporters)?

**Comment by ReagentX** [comment]
I'm happy to work on this if there's enough interest, but we must think about what a feature-complete Markdown export would look like. Can you provide some examples?

**Reply by jawadakhan97** [reply]
hmm, yeah i wonder, since there aren't tables or like headings?

maybe a markdown file per convo, each date is a heading, and attachments are inserted using github-flavored markdown?

**Reply by ReagentX** [reply]
What about replies/threads, reactions, sticker types, edits, etc.?
